Ensure that the correct version of Qt is always used.

With the introduction qt-4.8.6, Qt binaries were moved from /usr/bin to
/usr/$(get_libdir)/qt4/bin, leaving behind in their place symlinks to qtchooser.

There is no guarantee to which version of Qt these symlinks might point, so it
is necessary to find the correct version explicitly.

Once qmake is found, it is queried for the correct location of all other items.

Gentoo-bug: 562746
Gentoo-bug: 583506

--- a/cmake/modules/FindQt4.cmake
+++ b/cmake/modules/FindQt4.cmake
@@ -446,6 +446,15 @@ GET_FILENAME_COMPONENT(qt_install_version "[HKEY_CURRENT_USER\\Software\\trollte
 # check for qmake
 # Debian uses qmake-qt4
 # macports' Qt uses qmake-mac
+if(CMAKE_BUILD_TYPE STREQUAL Gentoo)
+  find_program(QT_QMAKE_EXECUTABLE NAMES qmake
+    PATHS
+      $ENV{EPREFIX}/usr/${CMAKE_INSTALL_LIBDIR}/qt4/bin
+      $ENV{EPREFIX}/usr/bin
+      NO_DEFAULT_PATH
+    DOC "The qmake executable for the Qt installation to use"
+  )
+else()
 FIND_PROGRAM(QT_QMAKE_EXECUTABLE NAMES qmake qmake4 qmake-qt4 qmake-mac PATHS
   "[HKEY_CURRENT_USER\\Software\\Trolltech\\Qt3Versions\\4.0.0;InstallDir]/bin"
   "[HKEY_CURRENT_USER\\Software\\Trolltech\\Versions\\4.0.0;InstallDir]/bin"
@@ -453,6 +462,7 @@ FIND_PROGRAM(QT_QMAKE_EXECUTABLE NAMES qmake qmake4 qmake-qt4 qmake-mac PATHS
   $ENV{QTDIR}/bin
   DOC "The qmake executable for the Qt installation to use"
 )
+endif()
 
 IF (QT_QMAKE_EXECUTABLE)
 
